server.js

```diff
@@ -0,0 +1,98 @@
+const express = require('express');
+const app = express();
+const bodyParser = require('body-parser');
+const databaseHandler = require('./database.js');
+
+app.set('view engine', 'ejs');
+app.use(bodyParser.urlencoded({extended: true}));
+
+databaseHandler.setup( (err, res) => {
+	if(err) return console.log(err);
+	app.listen(3000, () => {
+	console.log('listening on port 3000');
+	console.log('any information inserted into this program will be removed when the program stops');
+	});	
+})
+
+
+app.get('/', (req, res) => {
+	res.sendFile(__dirname + '/views/index.html');
+});
+
+app.get('/insertForm.html', (req, res) => {
+	res.sendFile(__dirname + '/views/insertForm.html');
+});
+
+app.get('/updateForm.html', (req, res) => {
+	res.sendFile( __dirname + '/views/updateForm.html' );
+});
+
+app.get('/deleteForm.html', (req, res) => {
+	res.sendFile( __dirname + '/views/deleteForm.html');
+});
+
+app.get('/search.html', (req, res) => {
+	res.sendFile(__dirname +'/views/search.html');
+})
+
+app.get('/getAll', (req, res) => {
+	databaseHandler.getAllUsers( (err, result)  => {
+		if(err) return console.log(err);
+		res.render('viewInfo.ejs', {user_info: result});
+	})
+});
+
+//do a search by forename
+app.get('/search', (req, res) => {
+	//having _ as a default value being passed into the sql to say  Any single character
+	let params = {
+		forename: req.query.forename.toLowerCase() || '_',
+		surename: req.query.surename.toLowerCase() || '_'
```

Comment:
if the keys are missing it will use `_` as a search term which will tell it to bring back every thing, so if you dont pass anything into the function it will return all of the users in the database